### ISSUE TITLE
Debian 10 minimal does not boot with distro kernel

### ISSUE THREAD

**Issue by leinsburg** [QubesOS/qubes-issues#5733]
Qubes OS version: 
4.0.3

Affected component(s) or functionality: 
debian-10-minimal

Install debian-10-minimal template:
`[user@dom0 ~]$ sudo qubes-dom0-update --enablerepo=qubes-templates-itl-testing debian-10-minimal`

Follow https://www.qubes-os.org/doc/managing-vm-kernel/#installing-kernel-in-debian-vm

Set debian-10-minimal to use distro kernel, virt_mode to HVM:
`[user@dom0 ~]$ qvm-prefs debian-10-minimal kernel '' && qvm-prefs debian-10-minimal virt_mode HVM`

Boot debian-10-minimal.

Expected or desired behavior:
The template debian-10-minimal boots successfully.

Actual behavior:
The template debian-10-minimal fails to boot, with the message "Cannot connect to qrexec agent for 60 seconds, see /var/log/xen/console/guest-debian-10-minimal.log". I'm unsure if there's anything that needs to be redacted, so I won't be including the logfile for now.

General notes:
Encountered this issue while setting up https://github.com/tasket/Qubes-vpn-support/wiki/Wireguard-VPN-connections-in-Qubes-OS, but with the debian-10-minimal template instead. Since debian-10-minimal doesn't boot, I'm stuck at step 2. Related issue: #5490

**Comment by unman** [comment]
Works for me.
Did you follow the instructions in full?
In particular, creating /boot/grub, installing all required packages, and then installing grub to system drive? Boot off qubes-kernel and check all the above.

**Comment by marmarek** [comment]
This looks like https://github.com/QubesOS/qubes-issues/issues/5490
Make sure you have all the updates installed.

BTW for Wireguard kernel module you can also install kernel-latest-qubes-vm in dom0 (which includes wireguard by default) and set VM to 5.5.x kernel.

**Comment by neowutran** [comment]
I have the same issue. 

Debian 10
--------------

Followed https://www.qubes-os.org/doc/managing-vm-kernel/#installing-kernel-in-debian-vm

```bash
sudo mkdir -p /boot/grub
sudo apt install --no-install-recommends linux-image-amd64 linux-headers-amd64 grub2-common qubes-kernel-vm-support
sudo update-grub
```

Set HVM and no kernel. 
=> Work 

Debian 10 minimal
------------
Do the same things

=> Doesn't work ( VM seabios start, but unable to find a bootable thing / bootloader )




I ran into this issue while playing with GPUs and went "manual" trying multiple things: https://neowutran.ovh/qubeos.html#using-a-kernel-provided-by-debian
Later (aka: today) I ran some more test to try to find the minimal number of command to make it work on a Debian 10 minimal. I issue is indeed that the bootloader is not installed. 
Instead of following the docs and running: 

```bash
sudo mkdir -p /boot/grub
sudo apt install --no-install-recommends linux-image-amd64 linux-headers-amd64 grub2-common qubes-kernel-vm-support
sudo update-grub
```

Just run as root:

```bash
apt install linux-image-amd64 linux-headers-amd64 grub2 qubes-kernel-vm-support
```
And when the grub gui popup, select /dev/xvda. 
After that it work, at least for me.
